app/code/Codifi/CustomerRequest/Model/ExportCsv.php

```diff
@@ -0,0 +1,175 @@
+ * @package Codifi\CustomerRequest\Model
+ */
+class ExportCsv
+{
+    /**
+     * Filesystem
+     *
+     * @var Filesystem
+     */
+    protected $fileSystem;
+
+    /**
+     * Note repository
+     *
+     * @var NoteRepository
+     */
+    private $noteRepository;
+
+    /**
+     * Csv processor
+     *
+     * @var Csv
+     */
+    private $csvProcessor;
+
+    /**
+     * Directory list
+     *
+     * @var DirectoryList
+     */
+    private $directoryList;
+
+    /**
+     * Filter builder
+     *
+     * @var FilterBuilder
+     */
+    private $filterBuilder;
+
+    /**
+     * Search criteria builder
+     *
+     * @var SearchCriteriaBuilder
+     */
+    private $searchCriteriaBuilder;
+
+    /**
+     * TimezoneInterface
+     *
+     * @var TimezoneInterface
+     */
+    private $timeZoneInterface;
+
+    /**
+     * New directory
+     */
+    private $newDirectory;
+
+    /**
+     * ExportCsv constructor.
+     *
+     * @param Filesystem $fileSystem
+     * @param NoteRepository $noteRepository
+     * @param Csv $csvProcessor
+     * @param DirectoryList $directoryList
+     * @param FilterBuilder $filterBuilder
+     * @param SearchCriteriaBuilder $searchCriteriaBuilder
+     * @param TimezoneInterface $timeZoneInterface
+     * @throws FileSystemException
+     */
+    public function __construct(
+        Filesystem $fileSystem,
+        NoteRepository $noteRepository,
+        Csv $csvProcessor,
+        DirectoryList $directoryList,
+        FilterBuilder $filterBuilder,
+        SearchCriteriaBuilder $searchCriteriaBuilder,
+        TimezoneInterface $timeZoneInterface
+    ) {
+        $this->noteRepository = $noteRepository;
+        $this->csvProcessor = $csvProcessor;
+        $this->directoryList = $directoryList;
+        $this->filterBuilder = $filterBuilder;
+        $this->searchCriteriaBuilder = $searchCriteriaBuilder;
+        $this->timeZoneInterface = $timeZoneInterface->date();
+        $this->newDirectory = $fileSystem->getDirectoryWrite(DirectoryList::VAR_DIR);
+    }
+
+    /**
+     * Pack customer notes to archive and delete from table
+     *
+     * @param int $period
+     * @return string
+     * @throws NoSuchEntityException
+     */
+    public function export(int $period): string
+    {
+        $currentDateTime = $this->timeZoneInterface->format('Y_m_d H:m:s');
+
+        $date = strtotime($currentDateTime . "-1 $period");
+        $newDate = date('Y-m-d H:m:s', $date);
+
+        $updatedAt = $newDate;
+
+        $this->filterBuilder->setField('updated_at');
+        $this->filterBuilder->setConditionType('to');
+        $this->filterBuilder->setValue($updatedAt);
+        $filter = $this->filterBuilder->create();
+
+        $this->searchCriteriaBuilder->addFilter($filter);
+        $searchCriteria = $this->searchCriteriaBuilder->create();
+
+        $noteList = $this->noteRepository->getList($searchCriteria);
+
+        $noteListItems = $noteList->getItems();
+
+        $content[] = [
```

Comment:
For getting headers for csv file, you can use `array_keys` with first item in $noteListItems